sponsorship_switzerland/models/contract_group.py

```diff
@@ -101,6 +97,26 @@ def write(self, vals):
             invoices.env.clear()
             invoices.write(inv_vals)
             invoices.action_invoice_open()
+
+        # When the payment mode changes to LSV or DD, we need to update the related open
+        # invoices (block above) and raise any errors first before changing the status
+        # of the contracts (block below). Otherwise if an error occurs, we have an
+        # undesirable situation where the status of the contracts are modified and the
+        # payment method did not successfully change.
+        if "payment_mode_id" in vals:
+            payment_mode = (
+                self.env["account.payment.mode"]
+                    .with_context(lang="en_US")
+                    .browse(vals["payment_mode_id"])
+            )
+            payment_name = payment_mode.name
+            for group in self:
+                old_term = group.payment_mode_id.name
```

Comment:
This cannot be right, because you moved the check after the `super` call. However when calling `super`, the new values are written. So the old_term will always be equal to the new term here. All your block should be done before calling `super`

Reply:
Corrected with commit 529ffdb